lib/new_relic/traced_thread.rb

```diff
@@ -0,0 +1,38 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+# frozen_string_literal: true
+
+module NewRelic
+  #
+  # This class creates a thread that contains instrumentation to allow the agent to see spans created inside of this thread.
```

Comment:
What about something like this?
```suggestion
  # This class allows the state of a transaction to be passed to a Thread so that nested segments can be created from the operations performed within the Thread's block. 
```